sdks/java/core/src/test/java/org/apache/beam/sdk/coders/CoderRegistryTest.java

```diff
@@ -89,8 +90,14 @@ public void testSerializableFallbackCoderProvider() throws Exception {
   public void testProtoCoderFallbackCoderProvider() throws Exception {
     CoderRegistry registry = getStandardRegistry();
 
+    //must use reflection here or a reference to dataflow class is baked into
```

Comment:
For me, they ARE running with the standard JDK compiler as well.  There are two tests in CoderRegistryTest that are marked with @Category(NeedsRunner.class) and thus should be run as part of the integration-tests.   The rest of the tests in CoderRegistryTest are not, just the two that are marked with the NeedsRunner.  

I can bump Tycho to 0.25, but then building with eclipse-jdt will require Java8.   I'm not opposed to that at all since it's main purpose would be for the Travis builds where we can mandate Java8, but I was trying to make sure we can use Java7 for now.